app_flutter/lib/service/appengine_cocoon.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:fixnum/fixnum.dart';
+
+import 'package:cocoon_service/protos.dart'
+    show Commit, CommitStatus, Stage, Task;
+
+import 'cocoon.dart';
+
+/// CocoonService for interacting with flutter/flutter production build data.
+///
+/// This queries API endpoints that are hosted on AppEngine.
+class AppEngineCocoonService implements CocoonService {
+  /// The Cocoon API endpoint to query
+  ///
+  /// This is the base for all API requests to cocoon
+  static const baseApiUrl = 'https://flutter-dashboard.appspot.com/api';
+
+  http.Client client = http.Client();
+
+  @override
+  Future<List<CommitStatus>> getStats() async {
+    /// This endpoint returns JSON [List<Agent>, List<CommitStatus>]
+    http.Response response = await client.get('$baseApiUrl/public/get-status');
+
+    if (response.statusCode != 200) {
+      throw new HttpException(
+          '$baseApiUrl/public/get-status returned ${response.statusCode}');
+    }
+
+    Map<String, dynamic> jsonResponse = jsonDecode(response.body);
+    assert(jsonResponse != null);
```

Comment:
I'm not sure if `jsonDecode` can actually return null. If it encounters an error I believe it throws a FormatException

Reply:
Okay, removed.